File: Java/src/main/java/Solution901_1000/Solution941.java
package Solution901_1000;

/**
 * @author windows
 */
public class Solution941 {
    public boolean validMountainArray(int[] arr) {
        int max = Integer.MIN_VALUE;
        int maxIndex = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > max){
                max = arr[i];
                maxIndex = i;
            }
        }
        if (maxIndex == 0 || maxIndex == arr.length - 1){
            return false;
        }
        for (int i = 0; i < maxIndex; i++) {
            if (arr[i] >= arr[i + 1]){
                return false;
            }
        }
        for (int i = maxIndex; i < arr.length - 1; i++) {
            if (arr[i] <= arr[i + 1]){
                return false;
            }
        }
        return true;
    }


    public static void main(String[] args) {
        Solution941 solution941 = new Solution941();
        System.out.println(solution941.validMountainArray(new int[]{2, 1}));
    }
}

File: C++/src/Solution901_1000/Solution941.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution941.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool validMountainArray(vector<int>& arr) {
  int left = 0, n = arr.size(), right = n - 1;
  while (left + 1 < n && arr[left] < arr[left + 1]){
      left++;
  }
  while (right - 1 >= 0 && arr[right] < arr[right - 1]){
      right--;
  }
  return left != 0 && right != n -1 && left == right;
}